src/repr/proto/adt/array.proto

```diff
@@ -0,0 +1,23 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+syntax = "proto3";
+
+package adt.array;
+
+message ProtoInvalidArrayError {
```

Comment:
I followed the convention in existing `*.proto` files (e.g. `row`). I am happy to use identical names (that is, without the `Proto` prefix), although this will make the `From` and `TryFrom` code a bit less readable.

Reply:
> IMO it's useful to have a reminder that the type you're dealing with cares very much about forward/backward compatibility.

For clients this should not be of concern. If you decide to change the type this should be evident, because the type is generated and the only way to change it is from the corresponding `*.proto` file. I'm not sure if this changes your opinion.